### ISSUE TITLE
Rebase from linguist again

### ISSUE THREAD

**Issue by mdespuits** [grosser/language_sniffer#2]
Any chance you could update from linguist again? There are some new features that would be very useful on the project I am working on. Thanks!

**Comment by grosser** [comment]
Took a look, but it's a mess and I also killed binary+vendor so merge/rebase is not really an option.

Most important changes are here:
https://github.com/grosser/language_sniffer/commit/09e1c96f47de880233a84ef0183bb1df6b090802

I think the safes/best way would be to checkout linguist master, reapply the changes and the force-push
- instead of deleting sections -> `=begin section =end` -> no conflicts when merging
- keep vendor/binary logic (not sure why I removed binary but I think it needed some kind of other dependency)

**Comment by mdespuits** [comment]
Yeah, I made a quick attempt, but as I am not familiar with this gem's internals at all yet, I just gave up for now. Technically, we're not doing any computation against the methods I mentioned like `binary?` and `image?`, so I may be able to avoid that. Yes, it did require `charlock_holmes`, which we are trying to eliminate as it is not only another gem dependency, but it also has a system level dependency...

For now, I will just see about avoiding that mess. If not, I'll refer back here.
